skiko/src/awtMain/kotlin/org/jetbrains/skiko/redrawer/MetalRedrawer.kt

```diff
@@ -154,13 +155,14 @@ internal class MetalRedrawer(
         windowOcclusionStateChannel.trySend(isOccluded)
     }
 
-    private fun performDraw() = synchronized(drawLock) {
+    private fun performDraw(waitVsync: Boolean = true) = synchronized(drawLock) {
         if (!isDisposed) {
-            // Wait for vsync because:
-            // - macOS drops the second/next drawables if they are sent in the same vsync
-            // - it makes frames consistent and limits FPS
-            displayLinkThrottler.waitVSync(windowPtr = layer.windowHandle)
-
+            if (waitVsync) {
+                // Wait for vsync because:
+                // - macOS drops the second/next drawables if they are sent in the same vsync
+                // - it makes frames consistent and limits FPS
+                displayLinkThrottler.waitVSync(windowPtr = layer.windowHandle)
```

Comment:
@elijah-semyonov, as far as I remember not calling this caused frame loosing, no matter what `maximumDrawableCount` was (a default is 3?)

Reply:
That's right.

Reply:
@SergeevPavel, you should take this into account. If you disable waiting for vsync on macOs, some of the last drawn frame can be discarded. The alternative solution to vsync is might be [this](https://github.com/JetBrains/skiko/pull/968/files#r1695524472).